internal/translation/searchindex/searchIndexsvc.go

```diff
@@ -0,0 +1,109 @@
+package searchindex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+var (
+	// ErrNotFound means an resource is missing
+	ErrNotFound = fmt.Errorf("not found")
+)
+
+type AtlasSearchIdxService interface {
+	GetIndex(ctx context.Context, projectID, clusterName, indexName, indexID string) (*SearchIndex, error)
+	CreateIndex(ctx context.Context, projectID, clusterName string, index *SearchIndex) (*SearchIndex, error)
+	DeleteIndex(ctx context.Context, projectID, clusterName, indexID string) error
+	UpdateIndex(ctx context.Context, projectID, clusterName string, index *SearchIndex) (*SearchIndex, error)
+}
+
+type ProductionAtlasSearch struct {
+	searchAPI admin.AtlasSearchApi
+}
+
+func NewAtlasDatabaseUsersService(ctx context.Context, provider atlas.Provider, secretRef *types.NamespacedName, log *zap.SugaredLogger) (*ProductionAtlasSearch, error) {
+	client, err := translation.NewVersionedClient(ctx, provider, secretRef, log)
+	if err != nil {
+		return nil, err
+	}
+	return NewProductionAtlasSearch(client.AtlasSearchApi), nil
+}
+
+func NewProductionAtlasSearch(api admin.AtlasSearchApi) *ProductionAtlasSearch {
+	return &ProductionAtlasSearch{searchAPI: api}
+}
+
+func (si *ProductionAtlasSearch) GetIndex(ctx context.Context, projectID, clusterName, indexName, indexID string) (*SearchIndex, error) {
+	resp, httpResp, err := si.searchAPI.GetAtlasSearchIndex(ctx, projectID, clusterName, indexID).Execute()
+	if err != nil {
+		if httpResp.StatusCode == http.StatusNotFound {
+			return nil, errors.Join(err, ErrNotFound)
+		}
+		return nil, err
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("received an empty index. Index: %s(%s). Status code: %d, E: %w",
+			indexName, indexID, httpResp.StatusCode, err)
+	}
+	stateInAtlas, err := fromAtlas(*resp)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert index to AKO. Index: %s(%s, E: %w",
+			indexName, indexID, err)
+	}
+	return stateInAtlas, nil
+}
+
+func (si *ProductionAtlasSearch) CreateIndex(ctx context.Context, projectID, clusterName string, index *SearchIndex) (*SearchIndex, error) {
+	atlasIndex, err := index.toAtlas()
+	if err != nil {
+		return nil, err
+	}
+	resp, httpResp, err := si.searchAPI.CreateAtlasSearchIndex(ctx, projectID, clusterName, atlasIndex).Execute()
+	if err != nil || httpResp.StatusCode != http.StatusCreated && httpResp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to create index: %w, status: %d", err, httpResp.StatusCode)
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("returned an empty index as a result of creation")
```

Comment:
there is no formatting here, just
```suggestion
		return nil, errors.New("returned an empty index as a result of creation")
```

this should've been captured by linting 🤔

Reply:
done